packages/core/src/components/mediaDeviceSelect.ts

```diff
@@ -12,7 +12,11 @@ export type SetMediaDeviceOptions = {
   exact?: boolean;
 };
 
-export function setupDeviceSelector(kind: MediaDeviceKind, room?: Room) {
+export function setupDeviceSelector(
+  kind: MediaDeviceKind,
+  room?: Room,
+  localTrack?: LocalAudioTrack | LocalVideoTrack,
```

Comment:
by allowing a track to be passed in, we can switch devices on that track, which allows us to use client-sdk-js for the switching logic here as well